src/ring_theory/maps.lean

```diff
@@ -32,28 +32,56 @@ namespace ring_equiv
 
 open ring_equiv
 
-variables {R F} [ring R] [ring F] (Hs : R ≃+* F) (x y : R)
+variables {R F} [semiring R] [semiring F] (Hs : R ≃+* F)
 
 lemma bijective : function.bijective Hs :=
 Hs.to_equiv.bijective
 
-lemma map_zero_iff {x : R} : Hs x = 0 ↔ x = 0 :=
+lemma map_zero_iff (x : R) : Hs x = 0 ↔ x = 0 :=
 ⟨λ H, Hs.bijective.1 $ H.symm ▸ Hs.map_zero.symm,
 λ H, H.symm ▸ Hs.map_zero⟩
 
 end ring_equiv
 
 /-- A ring involution -/
-structure ring_invo [ring R] extends R ≃+* opposite R :=
-(to_fun_to_fun : ∀ x, (to_fun (to_fun x).unop).unop = x)
+structure ring_invo [semiring R] extends R ≃+* opposite R :=
```

Comment:
You could use superscript `op` here, right?

Reply:
Done.